### PR TITLE
Document for before filing a PR

### LINKED ISSUE #206: Doc: Before you file a pull request
This doc under **Creating Pull Requests** should cover what people need to do before filing a pull request:

* Adding compute files to S3 (link to AWS docs on uploading to their personal results bucket)
* Reporting software and compute requirements (link to docs on determining software requirements/version and determining compute requirements)

This is currently blocked by https://github.com/AlexsLemonade/OpenScPCA-admin/issues/126 and https://github.com/AlexsLemonade/OpenScPCA-admin/issues/116

### PR DESCRIPTION
**Pull request by allyhawkins**
### Which issue does this address?
Closes #206 

### Briefly describe the scope of the added docs file, including whether you link out to any external docs.

This adds a document outlining the steps that should be completed before filing your PR. I included the three things that were mentioned in the issue: 

- merging in main to your feature branch 
- uploading results files to S3
- determining requirements 

I mostly linked to instructions we have in other parts of the docs, but let me know if I should expand more in certain areas. 

I also added a tip to the doc on filing a PR to reference this document first. 

### Will you need additional visual aids for these docs?
No

### Any other comments for reviewers?

I also noticed the joining AWS doc was missing from navigation so I added it here. 

### Author checklist

- [x] The docs page has been added to the `nav` section in `mkdocs.yml`
- [x] The docs adhere to the style guide [**structure directions**](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/general-style-guide.md#structure)
- [x] The docs adhere to the style guide [**language directions**](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/general-style-guide.md#structure)
- [x] The docs adhere to the style guide [**tone directions**](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/general-style-guide.md#tone)
- [x] The docs follow additional contributing instructions in [the docs contributing file](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/contributing.md)

### Reviewer checklist

Please refer to the [docs style guide](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/general-style-guide.md) and the [docs contributing guide](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/contributing.md) while reviewing this pull request.

- [ ] Do the docs adhere to the style guide [**structure directions**](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/general-style-guide.md#structure)?
- [ ] Do the docs adhere to the style guide [**language directions**](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/general-style-guide.md#structure)?
- [ ] Do the docs adhere to the style guide [**tone directions**](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/general-style-guide.md#tone)?
- [ ] Do the docs follow additional contributing instructions in [the docs contributing file](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/docs/contributing.md)?

### CODE REVIEW

**Review by sjspielman — COMMENTED** [review]
This all looks good to me! 

We have a couple links about things to do before filing that right now point to a general PR location, or they don't have a link at all. Can we update them to point to this new page?

- `software-platforms/aws/index.md`: We also use S3 as central location for you to share your result files with us **before** [filing a pull request](../../contributing-to-analyses/creating-pull-requests/index.md).
- `contributing-to-analyses/creating-pull-requests/file-pull-request.md`:    If you do not see a green check mark with the words `Able to merge`, but instead see a red X and a warning stating `Merge conflicts detected`, you will need to [resolve merge conflicts](resolve-merge-conflicts.md) **before** you can file the PR.
- `contributing-to-analyses/creating-pull-requests/resolve-merge-conflicts.md`:_We strongly recommend doing this **before** you file a PR._ 
- `contributing-to-analyses/determining-requirements/determining-software-requirements.md`:You should perform this step **before** filing a pull request to ensure that the `conda-lock.yml` file is up-to-date with the current state of your environment.

Can you also add a quick sentence at the top of the S3 buckets docs page in the section `## Syncing your results to S3` that links here and say that you need to do the syncing before you file a PR? Thanks!

**Comment by allyhawkins** [comment]
@sjspielman I updated some of the links that you mentioned in your original comment, but I actually think for most of those it makes sense to leave the link to the index for creating PRs. That's where all the information lives about what a PR is and how they are going to use it in OpenScPCA. Linking to this doc for many of those lines you pointed out just sends them back to those docs or in the case of the merge conflicts, sends them to the same link on resolving merge conflicts.

File: docs/software-platforms/aws/working-with-s3-buckets.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Working with S3 buckets
 
 As an onboarded OpenScPCA contributor, you will be [assigned an AWS S3 results bucket](../../getting-started/accessing-resources/index.md#getting-access-to-aws).
-You will use this bucket to [sync your analysis results](#syncing-your-results-to-s3).
+You will use this bucket to [sync your analysis results](#syncing-your-results-to-s3) before [filing a pull request](../../contributing-to-analyses/creating-pull-requests/index.md).
```
Comment (sjspielman) [inline docs/software-platforms/aws/working-with-s3-buckets.md:4]: We can probably get rid of my overzealous link to the same page here...

```suggestion
You will sync your analysis results to your bucket [before filing a pull request](../../contributing-to-analyses/creating-pull-requests/index.md).
```

File: docs/contributing-to-analyses/creating-pull-requests/resolve-merge-conflicts.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ If their branch is merged into the `main` branch of the upstream `AlexsLemonade/
 
 To resolve merge conflicts, you need to merge the upstream `main` branch into your feature branch.
 If Git finds merge conflicts, you can then resolve them in GitKraken, as described below.
-_We strongly recommend doing this before you file a PR._ <!-- STUB_LINK: Add link to before you file -->
+_We strongly recommend doing this [before you file a PR](./before-file-pr.md)._
```
Comment (sjspielman) [inline docs/contributing-to-analyses/creating-pull-requests/resolve-merge-conflicts.md:29]: this was really the main one to deal with :tada:

**Review by sjspielman — COMMENTED** [review]
I trust your discretion on links, let's go! 🚀

**Review by sjspielman — APPROVED** [review]
meant approve!

**Pull request merged**